### PR TITLE
Add request.performance event

### PR DESCRIPTION
**Pull request by riywo**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

To measure the performance of each single WebDriver command,
this commit adds a new event `performance` to WebDriverRequest object.

It also propagates the event to upper level as `request.performance` event
so that the consumer can set event listeners for this event on the browser object.

### Background
We use WDIO for our automation but we'd like to measure the performance (availability, latency and request counts) against our backend WebDriver endpoints
because we sometime see huge difference of performance based on endpoints.

To measure the performance, we can use [`command` and `result` events](https://github.com/webdriverio/webdriverio/blob/main/packages/webdriver/src/command.ts#L92-L101) but it's not ideal
because it includes retries and also it's hard to detect failure since exceptions are thrown and `result` event won't be fired. 

By this change, we just need to record `request.performance` events to measure the performance.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by riywo** [comment]
Could anyone take a look? At least, could you approve to run workflows so that I can catch any sanity check issues?

**Review by christian-bromann — COMMENTED** [review]
Thanks for raising the PR 👍 

I like the idea of allowing users to measure the performance of their WebDriver backend. One request I have would be to add a bit documentation around this.

**Comment by riywo** [comment]
@christian-bromann Thanks, I'll add the events section to https://github.com/webdriverio/webdriverio/blob/main/website/docs/BrowserObject.md

In the meantime, could you investigate the GitHub Actions failure? It looks like unrelated to my change because other PR failed as well https://github.com/webdriverio/webdriverio/actions/runs/2227431736

**Comment by riywo** [comment]
@christian-bromann I added a short description. https://github.com/webdriverio/webdriverio/pull/8267/files#diff-0a3bbfb43e4cbe97aa2f080fd9a8ba362651332753e90bc0b1d6fb6bc754f0dc

**Comment by christian-bromann** [comment]
Can you update your branch? Something seems off with the dependencies.

**Comment by riywo** [comment]
@christian-bromann I rebased on top of the upstream `main` branch (https://github.com/webdriverio/webdriverio/commit/665f3d08685bc27ba2138d0470ca57d3c0b46721) but the workflow still failed https://github.com/autifyhq/webdriverio/actions/runs/2240736251

It looks the latest commit is still failing https://github.com/webdriverio/webdriverio/actions/runs/2240408191

**Comment by christian-bromann** [comment]
Yeah, can you try to update the package-lock?
```sh
$ rm -r node_modules
$ npm i
```

**Comment by riywo** [comment]
@christian-bromann Those commands didn't update `package-lock.json` on my environment.

The broken package is specified by a tarball url https://github.com/visionmedia/dox/tarball/master which looks like updated. So, I tried:

```
npm i -D https://github.com/visionmedia/dox/tarball/master
```

Then, removed the dependency from `package.json`. Here is the commit: https://github.com/webdriverio/webdriverio/pull/8267/commits/f50e77d2faae9e1d5142c7c8fe4847de905ca3f5

**Comment by riywo** [comment]
@christian-bromann It looks working in our repository! https://github.com/autifyhq/webdriverio/actions/runs/2242440314

**Comment by riywo** [comment]
@christian-bromann Should I investigate the cause of e2e failures? Or, could you merge this PR without investigation?

**Comment by christian-bromann** [comment]
Seems like an error with the `jest-circus` dependency:

```
 TypeError: Cannot read properties of undefined (reading 'enableGlobally')
```

This is likely unrelated to your PR, so I will go ahead and merge but need to figure this out before we make a release.

File: website/docs/BrowserObject.md [hunk 1]
```diff
@@ -210,3 +210,105 @@ describe('my test', () => {
...
+      "rejectUnauthorized": true,
+    },
+    "method": "GET",
+    "responseType": "json",
+    "retry": 0,
+    "searchParams": Object {},
+    "throwHttpErrors": false,
+    "timeout": undefined,
+    "url": "https://localhost:4445/timeout",
+  },
```
Comment (christian-bromann) [inline website/docs/BrowserObject.md:310]: Let's keep the request object short
```suggestion
  "request": Object { ... },
```

Comment (riywo) [reply website/docs/BrowserObject.md]: Thanks, fixed!

**Comment by riywo** [comment]
@christian-bromann Let me know if I need to address anything before merging.

**Review by christian-bromann — APPROVED** [review]
👍 LGTM

Thanks for the PR!

**Comment by riywo** [comment]
It looks released by v7.19.7! Will check it out! https://github.com/webdriverio/webdriverio/commit/ed5b7278190953d5e77d11083b59db2209ac99b5 (Just missing this PR in the release note?)

**Comment by christian-bromann** [comment]
@riywo uff sorry, didn't properly tagged this PR. I will manually add it.

**Pull request merged**